sequana_pipetools/sequana_manager.py

```diff
@@ -585,24 +595,58 @@ def _hash(url):
 
         os.makedirs(self.singularity_prefix, exist_ok=True)
 
+        count = 0
+        files_to_download = []
+
+        # define the URLs and the output filename. Also, remove urls that
+        # have already been downloaded. 
         for url in urls:
             name = _hash(url)
-
             outfile = f"{self.singularity_prefix}/{name}.simg"
             if os.path.exists(outfile):
                 logger.info(f"Found corresponding image of {url} in {outfile}")
             else:
-                logger.info(f"Downloading {url}")
-                try:
-                    response = requests.get(url)
-                    with open(outfile, "wb") as fout:
-                        fout.write(response.content)
-                except requests.ConnectionError:
-                    logger.critical(
-                        f"{url} could not be downloaded. Your pipeline will probably won't work if you use --use-singularity. Continue with other images"
-                    )
-                    pass
+                files_to_download.append((url, outfile, count))
+                count += 1
+                logger.info(f"Preparing {url} for download")
+
+        try:    # try an asynchrone downloads
+            #multiple_downloads(urls_to_download, names, positions)
+            multiple_downloads(files_to_download)
+        except Exception: #pragma: no cover
+            try:
+                response = requests.get(url)
+                with open(outfile, "wb") as fout:
+                    fout.write(response.content)
+            except requests.ConnectionError:
+                logger.critical(
+                    f"{url} could not be downloaded. Your pipeline will probably won't work if you use --use-singularity. Continue with other images"
+                )
+                pass
+
+
+def multiple_downloads(files_to_download):
+
+    @asyncio.coroutine
+    def download(url, name, position):
```

Comment:
C'est quoi la diff entre le decorator et faire une fonction async ?

Reply:
> @coroutine is a functional analogue for async def but it works in Python 3.4+ and utilizes yield from construction instead of await . For practical perspective just never use @coroutine if your Python is 3.5+.

Reply:
je ne sais pas mais sans créér la coroutine manuellement avec le décorateur, le download ne démarre pas ...
fait également dans un test independent et j'ai eu le meme comportement